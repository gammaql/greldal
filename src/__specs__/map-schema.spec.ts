import { mapSchema } from "..";
import { GraphQLString, graphql, GraphQLObjectType, printSchema } from "graphql";

test("External resolver mapping", async () => {
    const schema = mapSchema([
        {
            operationType: "query" as "query",
            name: "hello",
            fieldConfig: {
                type: GraphQLString,
                resolve() {
                    return "world";
                },
            },
        },
    ]);
    const r1 = await graphql(
        schema,
        `
            query {
                hello
            }
        `,
    );
    expect(r1.errors).not.toBeDefined();
    expect(r1).toMatchSnapshot();
});

test("Schema interceptor", () => {
    const schema = mapSchema(
        [
            {
                operationType: "query" as "query",
                name: "hello",
                fieldConfig: {
                    type: GraphQLString,
                    resolve() {
                        return "world";
                    },
                },
            },
        ],
        schemaConfig => {
            schemaConfig.subscription = new GraphQLObjectType({
                name: "Subscription",
                fields: {},
            });
            return schemaConfig;
        },
    );
    expect(printSchema(schema)).toMatchSnapshot();
});
