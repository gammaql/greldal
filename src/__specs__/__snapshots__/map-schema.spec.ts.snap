// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`External resolver mapping 1`] = `
Object {
  "data": Object {
    "hello": "world",
  },
}
`;

exports[`Schema interceptor 1`] = `
"schema {
  query: query
  subscription: Subscription
}

type query {
  hello: String
}

type Subscription {

}
"
`;
